stonesoup/dataassociator/tracktotrack.py

```diff
@@ -302,3 +302,43 @@ def associate_tracks(self, tracks_set, truth_set):
                     TimeRange(start_timestamp, end_timestamp)))
 
         return AssociationSet(associations)
+
+
+class TrackIDbased(TrackToTrackAssociator):
+    """Track ID based associator
+
+        Compares set of :class:`~.Track` objects to set of :class:`~.GroundTruth` objects,
+        each formed of a sequence of :class:`~.State` objects and returns an
+        :class:`~.Association` object for each time at which a the two :class:`~.State`
+        within the :class:`~.Track` and :class:`~.GroundTruthPath` are assessed to be associated.
+        Tracks are considered to be associated with the Ground Truth if the ID of the Track
+        is the same as the ID of the Ground Truth.
+        """
+
+    def associate_tracks(self, tracks_set, truths_set):
+        """Associate two sets of tracks together.
+
+               Parameters
+               ----------
+               tracks_set : list of :class:`~.Track` objects
+                   Tracks to associate to ground truths set
+               truths_set: list of :class:`~.GroundTruthPath` objects
+                   Ground truths to associate to tracks set
+
+               Returns
+               -------
+               AssociationSet
+                   Contains a set of :class:`~.Association` objects
+
+               """
+
+        associations = set()
+
+        for track in tracks_set:
+            for truth in truths_set:
+                if track.id == truth.id:
+                    associations.add(TimeRangeAssociation((track, truth),
```

Comment:
Is this a `TimeRangeAssociation`, or is it being used because we don't have any other type? In principle, one might have tracks with no timestamps in which case "TimeRange" seems meaningless (though I don't think it'll error).

Elsewhere 'TimeRangeAssociation` seems to be for tracks that are only associated at particular times, and not elsewhere. This associator actually ignores timesteps but does attach the complete truth time range, which might not be correct for the track. Suggest we find, or create a simpler `Association` object.

Reply:
Could make this an `AssociationPair`, assuming unique IDs.

Reply:
But maybe not, while there may be unique truth, the tracks could be discontinuous (like with occlusion), you'd still like to associate the fragments with the same ID to the correct truth. So recommend `Association` then I suppose.

Reply:
I think I used `TimeRangeAssociation` as I thought that was what you need for the SIAP metrics - on a second look I don't think that's true so I'll look at using `Association` instead.

Reply:
It may be okay to use the time range association, but just use the track times, rather than the truth times? Or even the overlapping time (`max(track[0].timestamp, truth.timestamp[0])` and `min(track[-1].timestamp, truth[-1].timestamp`)